doc/source/serve/deployment-graph.md

```diff
@@ -0,0 +1,565 @@
+
++++
+
+```python
+from ray import serve
+# We will later move Ray DAG related components out of experimental in later stable release
+from ray.experimental.dag.input_node import InputNode
+
+@serve.deployment
+async def preprocessor(input_data: str):
+    """Simple feature processing that converts str to int"""
+    time.sleep(0.1) # Manual delay for blocking computation
+    return int(input_data)
+
+@serve.deployment
+async def avg_preprocessor(input_data):
+    """Simple feature processing that returns average of input list as float."""
+    time.sleep(0.15) # Manual delay for blocking computation
+    return sum(input_data) / len(input_data)
+
+# DAG building
+with InputNode() as dag_input:
+    # Partial access of user input by index
+    preprocessed_1 = preprocessor.bind(dag_input[0])  
+    preprocessed_2 = avg_preprocessor.bind(dag_input[1])
+```
+
++++
+
+There are two new APIs used in the DAG building stage: `InputNode()` and `bind()`.
+
+### **```InputNode()```** : User input of the graph
+
+```InputNode``` is a special singleton in DAG building that's only relevant to it's runtime call behavior. Even though all decorated classes or functions can be reused in arbitrary way to facilitate DAG building where the root DAGNode forms the graph with its children, in each deployment graph there should be one and only one InputNode used.
+
+```InputNode``` value is fulfilled and replaced by user input at runtime, therefore it takes no argument when being constructed.
+
+```InputNode``` should only be created within a context manager to indicate there's one and only one `InputNode` for each graph, and will throw exception if directly created, or multiple is found.
+
+It's possible to access partial user input by index or key, if a node in the graph doesn't need the complete user input to run. Example: User submits `[Tensor_0, Tensor_1]` via http but a model node might only need to receive `Tensor_1` at index 1 rather than complete list to reduce intermediate data transfer.
+
+
+### **`bind(*args, **kwargs)`** : The graph building API
+
+Once called on supported Ray-decorated function or class (`@serve.deployment` is fully supported, `@ray.remote` will be soon), generates a `DeploymentNode` of type `DAGNode` that acts as the building block of graph building.
+
+
+In the example above, we can see we're using a context manager to build and bind user input:
+```python
+with InputNode() as dag_input:
+```
+
+Which can be used and accessed by index or key in downstream calls of .bind(), such as:
+```python
+preprocessed_1 = preprocessor.bind(dag_input[0])
+```
+
+This means we're creating a DAGNode IR called `preprocessed_1` in graph building by calling `.bind()` on a serve decorated function, where it executes the decorated deployment function `preprocessor` that takes the user input at index 0 at runtime.
```

Comment:
```suggestion
This creates a deployment node called `preprocessed_1` by calling `.bind()` on the decorated function. Once deployed, this will translate to a deployment that runs `preprocessor` on the input at runtime.
```